src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1671,6 +1618,24 @@ private static TypeSymbol GetParameterType(int argIndex, MemberAnalysisResult re
             // UNDONE: Otherwise if one member is a non-lifted operator and  the other is a lifted
             // operator, the non-lifted one is better.
 
+            // Not spec'd yet: if two members are static extension methods, the more specific type that they
+            // are extending wins (even though it appears nowhere in the signature).
+            if (m1.Member.IsInExtensionClass && m2.Member.IsInExtensionClass && !m1.Member.IsUnreducedExtensionMember && !m2.Member.IsUnreducedExtensionMember)
+            {
+                var m1type = m1.Member.ContainingType.ExtensionClassType;
+                var m2type = m2.Member.ContainingType.ExtensionClassType;
+                if (Conversions.HasIdentityOrImplicitReferenceConversion(m1type, m2type, ref useSiteDiagnostics) ||
+                    Conversions.HasIdentityOrImplicitReferenceConversion(m2type, m1type, ref useSiteDiagnostics))
```

Comment:
I have changed this to `BetterConversionTarget` and added a test for `object` and a user struct ambiguity for instance methods. However, I'm actually not entirely too sure on the semantics of what static methods should actually do - I don't recall any in-depth discussion on the design of ambiguity resolution for static members, and that should be thought out before we finalize on this exact behavior. (I believe the example of specific types between object and a value type cannot come up, since ambiguity only arises in situations with a dependency chain at least two long, which doesn't happen with structs. There are more complex examples, though, that might involve value/reference conversions)